### ISSUE TITLE
[FEATURE] add `limit` argument to `ArgillaTrainer`

### ISSUE THREAD

**Issue by plaguss** [recognai/rubrix#4137]
**Is your feature request related to a problem? Please describe.**
Add a `limit` argument to the `ArgillaTrainer`, to align with #4072

**Describe the solution you'd like**
```python
trainer = ArgillaTrainer(
   ...
   limit: Optional[int] = 50
)
```

**Describe alternatives you've considered**

**Additional context**
This should be done after the limit argument is available (after [4074](https://github.com/argilla-io/argilla/pull/4074) is merged).

**Comment by davidberenstein1957** [comment]
@plaguss don't forget the labels ;)
